src/extractor/extraction_containers.cpp

```diff
@@ -390,6 +390,8 @@ void ExtractionContainers::PrepareEdges(ScriptingEnvironment &scripting_environm
 
         weight = scripting_environment.ProcessSegment(
             edge_iterator->source_coordinate, target_coord, distance, weight);
+        duration = scripting_environment.ProcessSegment(
```

Comment:
This probably needs an interface change for `ProcessSegment` to return both a weight and duration value. Maybe we should adapt the same interface other `Process*` functions have: It takes a `ExtractionSegment` object and modifies that.